### PR TITLE
collect missing call edges

### PR DESCRIPTION
**Pull request by rvantonder**
A call signature (annotated access) is not created for some calls. When that is the case, record the call edge for a `Call` as revealed by the AST.

### CODE REVIEW

File: analysis/callGraph.ml [hunk 1]
```diff
@@ -65,22 +65,33 @@ let create ~environment ~source =
...
               } ->
-                add_call_edge call_graph caller callee
+                Some callee
             | _ ->
-                call_graph
+                None
           in
           Annotated.Access.create access
-          |> Annotated.Access.fold ~resolution ~initial:call_graph ~f:add_call_edge
+          |> Annotated.Access.fold ~resolution ~initial:None ~f:resolve_access
```
Comment (rvantonder) [inline analysis/callGraph.ml]: @dkgi, basically, I noticed that `Annotated.Access.fold` doesn't create a `Signature` for some types of calls (see test). Not sure if this is "as intended" or if there is a missing case that should be added in `Access.fold`. For now, the logic below will catch missing edges.

File: analysis/callGraph.ml [hunk 2]
```diff
@@ -65,22 +65,33 @@ let create ~environment ~source =
...
+          |> function
+          | Some callee -> add_call_edge call_graph caller callee
+          | None ->
```
Comment (rvantonder) [inline analysis/callGraph.ml]: Basically, if we didn't resolve a call signature, check if the access is a `Access.Call` and then record it. @fahndrich this solves the missing edges issue and possibly fixes the low fixpoint iteration issue.

File: analysis/callGraph.ml [hunk 3]
```diff
@@ -38,21 +38,21 @@ let create ~environment ~source =
...
+        let process_access call_graph access =
+          let add_call_edge call_graph caller callee =
+            Log.log
+              ~section:`CallGraph
+              "Adding call edge %a -> %a"
+              Access.pp caller
+              Access.pp callee;
+            let update_callees = function
+              | Some callees -> Access.Set.add callees callee
+              | None -> Access.Set.add Access.Set.empty callee
```
Comment (sinancepel) [inline analysis/callGraph.ml]: Just `Access.Set.singleton callee`?

File: analysis/callGraph.ml (hunk 1 above)
Comment (dkgi) [reply analysis/callGraph.ml]: Not sure I see what you mean in the tests. Are we missing those just in conditional tests? If so, that sounds wrong.

Sounds like this should be fixed in the access fold. Let me know how you want to proceed. Fine with landing this and adding a TODO.

Comment (rvantonder) [reply analysis/callGraph.ml]: No it's not specific to conditional test (so I've removed that). It's just a minimal example of where call signatures are not created. 

> Sounds like this should be fixed in the access fold.

Possibly right? As a very simple example:

```python
def foo():
    bar()
```

When I fold over the `bar` access and print each element (there's only one), I see the resolution is only:

```
DUMP element: (AnnotatedAccess.Element.Attribute
   { AnnotatedAccess.Element.attribute = bar;
     origin = (AnnotatedAccess.Element.Module ); defined = false })
```

> Let me know how you want to proceed. Fine with landing this and adding a TODO.

Think this is fine for now until you have a chance to look into whether this is intended. Either way, the tests are good. Apparently you can edit my PR (e.g., add a TODO with task number), but I have put a placeholder TODO anyway.

**Pull request closed without merging**